engines/drascula/graphics.cpp

```diff
@@ -80,8 +82,20 @@ void DrasculaEngine::moveCursor() {
 			color_abc(kColorRed);
 	} else if (!_menuScreen && _color != kColorLightGreen)
 		color_abc(kColorLightGreen);
-	if (_hasName && !_menuScreen)
+	if (_hasName && !_menuScreen) {
+		if (_previousSaid != textName) {
+			Common::TextToSpeechManager *ttsMan = g_system->getTextToSpeechManager();
+			if (ConfMan.getBool("tts_enabled") && ttsMan != nullptr) {
+				ttsMan->say(textName);
+			}
+
+			_previousSaid = textName;
+		}
+
 		centerText(textName, _mouseX, _mouseY);
+	}
+	else if (!_menuBar && !_menuScreen)
+		_previousSaid.clear();
```

Comment:
```suggestion
		_previousSaid.clear();

```

It is better to separate visually distinct code blocks